### PR TITLE
[WIP] Migration to V8 API

### PR DESCRIPTION
**Pull request by PCatinean**
First attempt to migrate l10n_ch modules to the new API syntax.Started with l10n_ch_base_bank

### CODE REVIEW

**Comment by nbessi** [comment]
Hello Paul,

Thanks for the PR.
It would be really nice to have only migrated addons out of `_unported_`.
This will allows early merge and ease collaboration.

Regards 

Nicolas

File: l10n_ch_base_bank/bank.py [hunk 1]
```diff
@@ -52,19 +52,18 @@ def _check_5_pos_postal_num(self, number):
...
-    }
+
+    ### Internal reference
+    code = fields.Char(string='Code', size=64)
+    ###Swiss unik bank identifier also use in IBAN number
+    clearing = fields.Char('Clearing number', size=64)
+    ### city of the bank
+    city = fields.Char(string='City', size=128)
+    ### ccp of the bank
+    ccp = fields.Char(string='CCP', size=64)
```
Comment (yvaucher) [inline l10n_ch_base_bank/bank.py]: I think the comments should comes as help parameters of those fields.

File: l10n_ch_bank/__openerp__.py [hunk 2]
```diff
@@ -38,6 +38,6 @@
     "depends": ["l10n_ch"],
     "data": ["bank.xml"],
     "update_xml": [],
```
Comment (yvaucher) [inline l10n_ch_bank/__openerp__.py]: update_xml to be removed

File: l10n_ch_base_bank/bank.py [hunk 3]
```diff
@@ -127,39 +128,37 @@ def name_search(self, cursor, uid, name, args=None, operator='ilike', context=No
...
-        'acc_number': fields.char('Account/IBAN Number', size=64, required=True),
-        'ccp': fields.related('bank', 'ccp', type='char', string='CCP',
-                              readonly=True),
-    }
-
-    def get_account_number(self, cursor, uid, bid, context=None):
+    bvr_adherent_num = fields.Char(
+        string='Bank BVR adherent number', size=11,
+        help=("Your Bank adherent number to be printed in references of your BVR."
+            "This is not a postal account number.")
```
Comment (yvaucher) [inline l10n_ch_base_bank/bank.py]: Indentation issue here

File: l10n_ch_base_bank/bank.py [hunk 4]
```diff
@@ -127,39 +128,37 @@ def name_search(self, cursor, uid, name, args=None, operator='ilike', context=No
...
             bid = bid[0]
-        current = self.browse(cursor, uid, bid, context=context)
-        if current.state not in ('bv', 'bvr'):
+        if self.state not in ('bv', 'bvr'):
             return current.acc_number
-        if current.bank and current.bank.ccp:
+        if self.bank and current.bank.ccp:
             return current.bank.ccp
         else:
             return current.acc_number
```
Comment (yvaucher) [inline l10n_ch_base_bank/bank.py]: some `current` left to be changed to `self`

File: l10n_ch_base_bank/__init__.py [hunk 5]
```diff
@@ -17,5 +17,5 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from . import bank
-from . import invoice
+import bank
```
Comment (yvaucher) [inline l10n_ch_base_bank/__init__.py]: Why removing the `from .` ?

File: l10n_ch_dta/wizard/__init__.py [hunk 6]
```diff
@@ -19,7 +19,7 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from . import create_dta
-from . import unicode2ascii
+import create_dta
```
Comment (nbessi) [inline l10n_ch_dta/wizard/__init__.py]: Please respect realtive import

File: l10n_ch/__init__.py [hunk 7]
```diff
@@ -20,6 +20,6 @@
 #
 ##############################################################################
 
-from . import account_wizard
+import account_wizard
```
Comment (nbessi) [inline l10n_ch/__init__.py]: Please keep relative import

File: l10n_ch_base_bank/__init__.py (hunk 5 above)
Comment (nbessi) [inline l10n_ch_base_bank/__init__.py]: Please keep relative import

File: l10n_ch_dta/wizard/__init__.py (hunk 6 above)
Comment (nbessi) [inline l10n_ch_dta/wizard/__init__.py]: Keep relative import

File: l10n_ch_payment_slip/company.py [hunk 8]
```diff
@@ -19,10 +19,10 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp.osv.orm import Model, fields
+from openerp.osv import fields, orm
 
 
-class ResCompany(Model):
+class ResCompany(orm.Model):
```
Comment (nbessi) [inline l10n_ch_payment_slip/company.py:25]: Guess not ported to new API yet

File: l10n_ch_payment_slip/wizard/__init__.py [hunk 9]
```diff
@@ -0,0 +1 @@
+import bvr_import
```
Comment (nbessi) [inline l10n_ch_payment_slip/wizard/__init__.py]: from . import bvr_import

File: l10n_ch_base_bank/bank.py (hunk 1 above)
Comment (bwrsandman) [inline l10n_ch_base_bank/bank.py]: Either way, the `###` format needs to go... It's not pep8

File: l10n_ch_dta/wizard/__init__.py [hunk 10]
```diff
@@ -19,7 +19,7 @@
...
+import unicode2ascii
```
Comment (bwrsandman) [inline l10n_ch_dta/wizard/__init__.py]: This library needs to be defined in `__openerp__.py` as an external_dependency

File: l10n_ch_payment_slip/wizard/bvr_import.py [hunk 11]
```diff
@@ -24,14 +24,14 @@
 import re
 
 from openerp.tools.translate import _
-from openerp.osv.orm import TransientModel, fields
+from openerp.osv import fields, orm
 from openerp.osv.osv import except_osv
```
Comment (bwrsandman) [inline l10n_ch_payment_slip/wizard/bvr_import.py]: Most uses of `except_osv` should be replaced with `except_orm`

File: l10n_ch_payment_slip/company.py (hunk 8 above)
Comment (PCatinean) [inline l10n_ch_payment_slip/company.py:25]: Have no yet started proper migration of the module yet, soon to come though

File: l10n_ch_payment_slip/wizard/bvr_import.py (hunk 11 above)
Comment (PCatinean) [inline l10n_ch_payment_slip/wizard/bvr_import.py]: Have no yet started proper migration of the module yet, soon to come though

**Comment by PCatinean** [comment]
All the suggestions were warmly welcomed and listened to.I have quite a lot to learn so any tip ("Commit messages, syntax, pepping, style etc etc") is warmly welcomed and taken heed to

File: __unported__/l10n_ch/__openerp__.py [hunk 12]
```diff
@@ -58,7 +58,7 @@
  'demo': [],
  'test': [],
  'auto_install': False,
- 'installable': False,
+ 'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch/__openerp__.py:74]: if the addon is in **unported** please leave installable False.

File: __unported__/l10n_ch_bank/__openerp__.py [hunk 13]
```diff
@@ -38,6 +38,6 @@
     "depends": ["l10n_ch"],
     "data": ["bank.xml"],
     "update_xml": [],
-    "active": False,
-    'installable': False,
+    "active": True,
+    'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_bank/__openerp__.py]: ditto

File: __unported__/l10n_ch_credit_control_payment_slip_report/__openerp__.py [hunk 14]
```diff
@@ -34,6 +34,6 @@
                             "credit_control_printer_view.xml",
                             "report.xml"
                         ],
-        "active": False,
-        'installable': False
+        "active": True,
+        'installable': True
```
Comment (gurneyalex) [inline __unported__/l10n_ch_credit_control_payment_slip_report/__openerp__.py]: ditto

File: __unported__/l10n_ch_dta/__openerp__.py [hunk 15]
```diff
@@ -35,7 +35,7 @@
  'demo': ["demo/dta_demo.xml"],
  'test': [], # To be ported or migrate to unit tests or scenarios
  'auto_install': False,
- 'installable': False,
+ 'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_dta/__openerp__.py]: ditto

File: __unported__/l10n_ch_dta_base_transaction_id/__openerp__.py [hunk 16]
```diff
@@ -43,6 +43,6 @@
  'website': 'http://www.camptocamp.com',
  'data': [],
  'tests': [],
- 'installable': False,
- 'auto_install': True,
+ 'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_dta_base_transaction_id/__openerp__.py]: ditto

File: __unported__/l10n_ch_payment_slip_account_statement_base_completion/__openerp__.py [hunk 17]
```diff
@@ -47,6 +47,6 @@
  'data': ['data.xml',
           ],
  'tests': [],
- 'installable': False,
- 'auto_install': True,
+ 'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_payment_slip_account_statement_base_completion/__openerp__.py]: you know what I mean...

File: __unported__/l10n_ch_payment_slip_base_transaction_id/__openerp__.py [hunk 18]
```diff
@@ -43,6 +43,6 @@
  'website': 'http://www.camptocamp.com',
  'data': [],
  'tests': [],
- 'installable': False,
- 'auto_install': True,
+ 'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_payment_slip_base_transaction_id/__openerp__.py]: should be False

File: __unported__/l10n_ch_scan_bvr/__openerp__.py [hunk 19]
```diff
@@ -45,5 +45,5 @@
  "data": ["wizard/scan_bvr_view.xml",
           "partner_view.xml",
           "bank_view.xml"],
- 'installable': False
+ 'installable': True
```
Comment (gurneyalex) [inline __unported__/l10n_ch_scan_bvr/__openerp__.py]: should be False

File: __unported__/l10n_ch_sepa/__openerp__.py [hunk 20]
```diff
@@ -52,7 +52,7 @@
         "test/pain001_eu.yml",
         "test/pain001_ch.yml",
     ],
-    'installable': False,
+    'installable': True,
```
Comment (gurneyalex) [inline __unported__/l10n_ch_sepa/__openerp__.py]: should be False

**Comment by PCatinean** [comment]
@nbessi @yvaucher or anyone viewing the PR, is there anything else missing?

**Comment by nbessi** [comment]
@PCatinean seem good to me probably some PEP8  are left.
Adding some test will be also really nice but else it seems fine.

I will create a PR today  with meta files for localisation and upgrade branch 8.0 with last version seven fixes.
Once rebased on it, we should be fine

Nicolas

**Comment by nbessi** [comment]
@PCatinean the PR is here https://github.com/OCA/l10n-switzerland/pull/52

**Comment by PCatinean** [comment]
@nbessi tried to rebase and pep as best as I could, let me know if I missed anything or should have approched it differently

**Comment by nbessi** [comment]
hello @PCatinean , 

Thanks for the work,

Your rebase is incorrect there is files that should not be here.
I tries to rebase your branches too here

https://github.com/nbessi/l10n-switzerland/tree/fix_rebase

I have a correct result (I think...) but the diff will be probably too big to be accepted.
You work on a lots of the files in various addons so it makes rebasing complex.
I propose to start a new PR based on update_80 branch containing only base bank (We should not be picky if we loose some commit messages).

My apologies for the inconvenience and be sure that your work is really appreciated here :smile_cat:  

Regards

Nicolas

**Pull request closed without merging**